ui/src/components/layout/layouting.ts

```diff
@@ -0,0 +1,66 @@
+import dagre from "@dagrejs/dagre"
+import { Edge, Position } from "reactflow"
+import { EipFlowNode, Layout } from "../../api/flow"
+
+
+const DEFAULT_NODE_WIDTH = 128
+const DEFAULT_NODE_HEIGHT = 128
+
+const graph = new dagre.graphlib.Graph()
```

Comment:
While testing I noticed that having the graph as a module variable does not remove any deleted nodes, leading to "shadow" nodes accumulating in the graph's internal representation. We should move this constructor call to the body of `newFlowLayout`, giving us a clean-slate every time it is called.